domain/consensus/utils/serialization/common.go

```diff
@@ -188,5 +188,5 @@ func ReadElements(r io.Reader, elements ...interface{}) error {
 
 // IsMalformedError returns whether the error indicates a malformed data source
 func IsMalformedError(err error) bool {
-	return errors.Is(err, io.ErrUnexpectedEOF) || errors.Is(err, errMalformed)
+	return errors.Is(err, io.ErrUnexpectedEOF) || errors.Is(err, io.EOF) || errors.Is(err, errMalformed)
```

Comment:
A bit risky to consider EOF as Malformed. Not sure there's anything to do about it though, but please give it a thought.